### DISCUSSION TITLE
What are you using LibCST for?

### DISCUSSION THREAD

**Discussion by zsol** [Instagram/LibCST#687] [Show and tell]
Reading about all the different use cases helps keep me motivated, and we might learn from each other.

I'll start with an unusual one: @lpetre and I maintain an indexer that uses [ScopeProvider](https://libcst.readthedocs.io/en/latest/metadata.html#scope-metadata) to infer cross-references between Python files. This powers automatic dead code removal, and IDE features (like jump-to-def, find-references).

What are _you_ using LibCST for?

**Comment by Kludex** [comment]
Mainly to improve FastAPI ecosystem.

This week we merged https://github.com/encode/starlette/pull/1376 on Starlette, and I'm creating a codemod to make our users' life easier.

See: https://github.com/encode/starlette/pull/1376#pullrequestreview-1096740606

**Comment by Kludex** [comment]
Now I'm using to create `bump-pydantic` to help on the migration from v1 to v2. 🙏 

Ref.: https://github.com/pydantic/bump-pydantic

**Comment by Josverl** [comment]
Libcst is used in the [micropython-stubber](https://github.com/josverl/micropython-stubber) project to merge different partial stub definitions to create [stub packages](https://github.com/josverl/micropython-stubs) that are published to pypi, and to generate different variations of stub generation scripts that are run on micropython microcontrollers.
